packages/wrangler/src/__tests__/event-subscriptions.test.ts

```diff
@@ -0,0 +1,334 @@
+						return HttpResponse.json({
+							success: true,
+							errors: [],
+							result: [
+								{
+									id: "vitestResourceless",
+									events: [
+										{ name: "event-1", description: "Test event 1" },
+										{ name: "event-2", description: "Test event 2" },
+									],
+								},
+							],
+						});
+					},
+					{ once: true }
+				)
+			);
+
+			const result = runWrangler(
+				"event-subscriptions browse --source=vitestResourceless"
+			);
+
+			await expect(result).resolves.toBeUndefined();
+			expect(std.out).toMatchSnapshot();
+		});
+	});
+
+	describe("event-subscription create", async () => {
+		test("creates a new subscription", async () => {
+			const result = runWrangler(
+				"event-subscriptions create test-subscription --source=vitest.testName --destination=queueId --events=e1,e2"
+			);
+
+			await expect(result).resolves.toBeUndefined();
+			expect(std.out).toMatchSnapshot();
+		});
+
+		test("errors for unrecognized products", async () => {
+			const result = runWrangler(
+				"event-subscriptions create test-subscription --source=unrecognized.resourceId --destination=queueId --events=e1,e2"
+			);
+
+			await expect(result).rejects.toMatchSnapshot();
+		});
+
+		test("allows resource-less products without a resource ID", async () => {
+			const result = runWrangler(
+				"event-subscriptions create test-subscription --source=vitestResourceless --destination=queueId --events=e1,e2"
+			);
+
+			await expect(result).resolves.toBeUndefined();
+			expect(std.out).toMatchSnapshot();
+		});
+
+		test("errors when resource ID is missing", async () => {
+			const result = runWrangler(
+				"event-subscriptions create test-subscription --source=vitest --destination=queueId --events=e1,e2"
+			);
+
+			await expect(result).rejects.toMatchSnapshot();
+		});
+	});
+
+	test("event-subscriptions list", async () => {
+		msw.use(
+			http.get(
+				"*/accounts/:accountId/event_subscriptions/subscriptions",
+				async ({ params }) => {
+					expect(params.accountId).toEqual(testAccountId);
+					return HttpResponse.json({
+						success: true,
+						errors: [],
+						result: [
+							{
+								id: "test-subscription-id",
+								name: "test-subscription",
+								product_id: "test-product-id",
+								resource_id: "test-resource-id",
+								queue_id: "test-queue-id",
+								queue_name: "test-queue-id",
+								events: ["e1", "e2"],
+							},
+							{
+								id: "test-subscription-id-2",
+								name: "test-subscription-2",
+								product_id: "test-product-id-2",
+								resource_id: "test-resource-id-2",
+								queue_id: "test-queue-id-2",
+								queue_name: "test-queue-id-2",
+								events: ["e3", "e4"],
+							},
+						],
+					});
+				},
+				{ once: true }
+			)
+		);
+
+		const result = runWrangler("event-subscriptions list");
+		await expect(result).resolves.toBeUndefined();
+		expect(std.out).toMatchSnapshot();
+	});
+
+	test("event-subscriptions get", async () => {
```

Comment:
would be great if `get`/`update`/`delete` tests would test what happens when the provided `id` does not exist